schema/object.go

```diff
@@ -126,6 +132,26 @@ func (o *ObjectSchema) Unserialize(data any) (result any, err error) {
 	return rawData, nil
 }
 
+func (o *ObjectSchema) unserializeInlinedDataToMap(data any) (map[string]any, error) {
+	if len(o.Properties()) != 1 {
+		panic(fmt.Errorf("unserializeInlinedDataToMap called on ObjectSchema with %d"+
+			" properties; only 1 allowed", len(o.Properties())))
+	}
```

Comment:
Now your `panic` at the end is redundant -- `0` will get here as well as `>1`.

I'll approve anyway, but I wouldn't mind if you felt inclined to clean that up. (Mind you, the message there looks a bit nicer, since "with 0 properties; only 1 allowed" sounds a bit odd, if not strictly incorrect! Maybe "exactly 1 required" would be better phrasing for both cases.)

Reply:
Good point. Fixed.